ci/docker/runtime_functions.sh

```diff
@@ -1487,26 +1512,30 @@ nightly_test_imagenet_inference() {
 #Runs a simple MNIST training example
 nightly_test_image_classification() {
     set -ex
+    export DMLC_LOG_STACK_TRACE_DEPTH=10
```

Comment:
could you explain why do we need this everywhere? If you are so sold about 10, why not set it as a default somewhere? We have too many knobs.

Reply:
10 has been the default number before. Since we want to turn off stacktrace logging for end user, preserve existing defaults in dmlc-core, and still return useful debug information in CI when error occurs, it's most straightforward to treat CI environment as dev setup and add this environment variable to the CI test environments.